packages/backend/src/modules/DiscoveryModule.ts

```diff
@@ -154,6 +155,30 @@ export function createDiscoverySubmodule(
     logger,
   )
 
+  const cacheInvalidationIndexer = new CacheInvalidationIndexer(
+    repositories.blockNumber,
+    repositories.providerCache,
+    repositories.indexerState,
+    chainId,
+    logger,
+    blockNumberIndexer,
+  )
+
+  /**
+   * Do we want both indexers here?
+   * Waterfall update should be sufficient
+   *
+   *  Currently:
+   *
+   *  BlockNumberIndexer -> CacheInvalidationIndexer -> DiscoveryIndexer
+   *
+   *
+   *  In question:
+   *                     ______________________________
+   *                   /                                \
+   *                  /                                  v
+   *  BlockNumberIndexer -> CacheInvalidationIndexer -> DiscoveryIndexer
```

Comment:
Any of them works, on the second example Discovery Indexer has to wait for CacheInvalidationIndexer anyway, so let's keep it as is.

Reply:
https://github.com/l2beat/lz-monitoring/pull/37/commits/b19fc71a9f41fb5b15cddde390d56765f757766a